src/NServiceBus.Transport.RabbitMQ.TransportTests/ConfigureRabbitMQTransportInfrastructure.cs

```diff
@@ -88,14 +82,32 @@ ConnectionFactory CreateConnectionFactory()
     {
         var connectionFactory = new ConnectionFactory
         {
-            UserName = connectionStringBuilder["username"].ToString(),
-            Password = connectionStringBuilder["password"].ToString(),
-            VirtualHost = connectionStringBuilder["virtualhost"].ToString(),
-            HostName = connectionStringBuilder["host"].ToString(),
             AutomaticRecoveryEnabled = true,
             UseBackgroundThreadsForIO = true
         };
 
+        object value;
```

Comment:
Avoiding the NRE is good, but I also think from a general correctness standpoint, we shouldn't be trying to execute the cleanup purging if we failed in setup and didn't actually run a test, so 👍 for changing behavior here.